factcast-site/documentation-docsy/content/en/UI/Setup/plugins.md

```diff
@@ -0,0 +1,34 @@
+---
+title: "Plugins"
+type: docs
+weight: 200
+description: Plugins to enrich visual representation of data
+---
+
+## Plugins
+
+FactCast Server UI is extensible using Plugins. Those plugins can be handy for enriching the plain fact information you
+see.
+
+### Scenario
+
+Consider you have the following Events:
+
+- UserRegistered
+- UserLogin
+
+where UserRegistered has a UserId and a UserName, and UserLogin is emitted after a succesful login, having just the
+UserId.
+
+Now, when you look at UserLogin Events through the UI, the UserId is all you see, because thats all the json there is to
+your fact payload. However it might be nice to know the UserName for a UserId when you look at it. Also it would be
+cool, to 'annotate' the Json with that information. This is what Plugins can do:
+In the client, you'd know what to do: Build a projection that consumes all the UserRegistered Events to provide a
+queryable Map<UserId, UserName>.
+
+If you now do this in a Plugin, it gives you an API to inspect and annotate the Json that will be displayed to the User.
```

Comment:
```suggestion
Based on this map you can now add a Plugin that provides an API to inspect and annotate the Json that is displayed to the User and adds the UserNames to every event.
```

Reply:
same then above, will try to describe it differently.